actinia_stac_plugin/core/stac.py

```diff
@@ -0,0 +1,265 @@
+        elif not root_validation:
+            msg["Error-root"] = {"message": "Please check the URL provided (Should be a STAC Catalog) as well as the name given (no spaces or undercore characters)."}
+        elif not collection_validation:
+            msg["Error-collection"] = {"message": "Please check the URL provided (Should be a STAC Catalog)."}
+        elif not instance_validation:
+            msg["Error-instance"] = {"message": "Please check the ID given (no spaces or undercore characters)."}
+        
+        return msg
+    else:
+        return {"message": "The JSON give does not have either stac-instance-id or stac-collection-id or stac-url . Check the parameters provided"}
+
+def callStacCollection(collection: str):
+    try:
+        stac = redis_actinia_interface.read(collection)
+    except:
+        stac = {"Error":"Something went wrong, please check the collection to retrived"}
+    
+    return stac
+
+def callStacCatalog(collection: str, catalog: str):
+    try:
+        stac_dict = redis_actinia_interface.read(collection)
+        stac_root_url = stac_dict[catalog]['root']
+        response = requests.get(stac_root_url)
+        stac = response.content
+    except:
+        stac = {"Error":"Something went wrong, please check the collection and catalog to retrived"}
+    
+    return stac
+
+def deleteStac(json):
+    stac_instance_id = 'stac-instance-id' in json
+    stac_collecion_id = 'stac-collection-id' in json
+
+    if stac_instance_id and stac_collecion_id:
+        return deleteStacCollection(json['stac-instance-id'],json['stac-collection-id'])
+    elif not stac_collecion_id and stac_instance_id:
+        return deleteStacInstance(json['stac-instance-id'])
+    elif not stac_instance_id and stac_collecion_id:
+        return {"Error":"The parameter stac-instance-id is required"}
+    else:
+        return {"Error":"The parameter does not match stac-instance-id or stac-collection-id"}
+
+def deleteStacCollection(stac_instance_id:str, stac_collection_id : str):
+    redis_actinia = connectRedis()
+    try:
+        stac_instance = redis_actinia_interface.read(stac_instance_id)
+        del stac_instance[stac_collection_id]
+        redis_actinia_interface.update(stac_instance_id,stac_instance)
+    except:
+        return {"Error": "Please check that the parameters given are well typed and exist"}
+
+    return redis_actinia_interface.read(stac_instance_id)
+
+def deleteStacInstance(stac_instance_id:str):
+    redis_actinia = connectRedis()
+    try:
+        redis_actinia_interface.delete(stac_instance_id)
+        instances = redis_actinia_interface.read("stac_instances")
+        del instances[stac_instance_id]
+        redis_actinia_interface.update("stac_instances",instances)
+    except:
+        return {"Error": "Something went wrong please that the element is well typed"}
+    return {"message": "The instance --"+ stac_instance_id + "-- was deleted with all the collections stored inside"}
```

Comment:
To this python file applies the same as for the api/stac.py: it would be better to split it into multiple files, one for each endpoint (e.g. stac.py and stacCollections.py) If code is used by both endpoints, a third file would be great (e.g. common.py).